src/main/java/org/systemexception/lunarlander/main/TestJbox.java

```diff
@@ -0,0 +1,140 @@
+package org.systemexception.lunarlander.main;
+
+import org.jbox2d.collision.shapes.PolygonShape;
+import org.jbox2d.common.Vec2;
+import org.jbox2d.dynamics.*;
+import org.lwjgl.LWJGLException;
+import org.lwjgl.input.Keyboard;
+import org.lwjgl.input.Mouse;
+import org.lwjgl.opengl.Display;
+import org.lwjgl.opengl.DisplayMode;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.lwjgl.opengl.GL11.*;
+
+/**
+ * @author leo
+ * @date 29/12/15 14:22
+ */
+public class TestJbox {
+
+	private static final String WINDOW_TITLE = "Physics in 2D!";
+	private static final int[] WINDOW_DIMENSIONS = {640, 480};
+
+	private static final World world = new World(new Vec2(0, -9.8f));
+	private static final Set<Body> bodies = new HashSet<Body>();
+
+	private static void render() {
+		glClear(GL_COLOR_BUFFER_BIT);
+		for (Body body : bodies) {
+			if (body.getType() == BodyType.DYNAMIC) {
+				glPushMatrix();
+				Vec2 bodyPosition = body.getPosition().mul(30);
+				glTranslatef(bodyPosition.x, bodyPosition.y, 0);
+				glRotated(Math.toDegrees(body.getAngle()), 0, 0, 1);
+				glRectf(-0.75f * 30, -0.75f * 30, 0.75f * 30, 0.75f * 30);
+				glPopMatrix();
+			}
+		}
+	}
+
+	private static void logic() {
+		world.step(1 / 60f, 8, 3);
+	}
+
+	private static void input() {
+		for (Body body : bodies) {
+			if (body.getType() == BodyType.DYNAMIC) {
+				if (Keyboard.isKeyDown(Keyboard.KEY_A)) {
+					body.applyAngularImpulse(+0.01f);
+				} else if (Keyboard.isKeyDown(Keyboard.KEY_D)) {
+					body.applyAngularImpulse(-0.01f);
+				}
+				if (Keyboard.isKeyDown(Keyboard.KEY_W)) {
+					body.applyLinearImpulse(new Vec2(0, 0.05f), body.getPosition());
+				}
+				if (Mouse.isButtonDown(0)) {
+					Vec2 mousePosition = new Vec2(Mouse.getX(), Mouse.getY()).mul(0.5f).mul(1 / 30f);
+					Vec2 bodyPosition = body.getPosition();
+					Vec2 force = mousePosition.sub(bodyPosition);
+					body.applyForce(force, body.getPosition());
+				}
+			}
+		}
+	}
+
+	private static void cleanUp(boolean asCrash) {
+		Display.destroy();
+		System.exit(asCrash ? 1 : 0);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [System.exit() should not be used in J2EE/JEE apps](https://www.codacy.com/app/leo_4/lunar-lander/file/1684949575/issues/source?bid=3040650&fileBranchId=3041914#l70)